docs/dev/contributing.md

```diff
@@ -89,11 +89,27 @@ If your commit is only shipping documentation changes or example files, and is a
 complete no-op for the test suite, please add **[skip ci]** in the commit message
 body to skip the build and give that slot to someone else who does need it.
 
-### Squash your commits
+### Pull request workflow
 
-Please rebase your changes on `main` and squash your commits whenever possible,
-it keeps history cleaner and it's easier to revert things. It also makes developers
-happier!
+The goals ordered by priority are:
+- Make PR reviews (both initial and follow-up reviews) easy for reviewers using GitHub
+- On the `main` branch, have a meaningful commit history that allows understanding (even years later) what each commit does, and why. 
+
+#### Before the first in-depth PR review
+
+Before the first in-depth PR review, meaningful commits are best: logically-encapsulated commits help the reviews go quicker and make the job for the reviewer easier. Conflicts with `main` can be resolved with a `git rebase origin/main` and a force push if it makes the review easier.
+
+#### After the first in-depth review
+
+After the first in-depth review, to make follow-up reviews easier:
+- Avoid force pushes (in particular rewriting the history that was already reviewed, which makes follow up reviews painful)
```

Comment:
I think we should explain why and how it's linked to GH. People familiar with Git but not GH might find it weird otherwise.

WDYT:
```suggestion
- Avoid force pushes. In particular rewriting the history that was already reviewed, which makes follow up reviews painful as GitHub loose track of each comments.
```

Reply:
Small edit: "loses" instead of "loose"